core/src/main/java/org/lflang/analyses/statespace/StateSpaceDiagram.java

```diff
@@ -0,0 +1,217 @@
+/** A directed graph representing the state space of an LF program. */
+package org.lflang.analyses.statespace;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.lflang.TimeValue;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.generator.ReactionInstance;
+import org.lflang.graph.DirectedGraph;
+
+// FIXME: Use a linkedlist instead.
```

Comment:
I think this is fine as-is because the Java standard `LinkedList` is extremely limited in that it only allows linear structures without looping nor branching. But this seems to be encoding structures that can be cyclic.